hr_contract_amendment/models/hr_contract_amendment.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+#   Copyright (C) 2018 EBII (https://www.saaslys.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models, _
+
+
+class HrContractAmendmentChanges(models.Model):
+    _name = 'hr.contract.amendment.changes'
+    _description = 'HR contract amendment change'
+
+    amendment_id = fields.Many2one(
+        comodel_name='hr.contract.amendment', string="Amendment",
+        required=True, ondelete="cascade")
+    value_name = fields.Char(string="Contract attribute name")
+    value_before = fields.Char(string="Value before change")
+    value_new = fields.Char(string="New value")
+
+
+class HrContractAmendment(models.Model):
+    _name = 'hr.contract.amendment'
+    _description = 'HR contract amendment'
+
+    contract_id = fields.Many2one(
+        comodel_name='hr.contract', string="Referenced Contract",
+        required=True)
+    name = fields.Char(string="Amendment name")
+    date = fields.Date(required=True, default=fields.Date.context_today)
+    user_id = fields.Many2one('res.users', string='Registered by',
+                              default=lambda self: self.env.user)
+    amendment_type = fields.Selection(selection=[
```

Comment:
Design question. Generally, in a amendment, it is possible to change many things in a same time. (salary, working hours, etc...)
a Many2Many implementation will suit better to the use case, don't you think ?

In france, an amendment (avenant au contrat) can belong severals changes.

Reply:
I agree with @legalsylvain . Amendment should cover change for several changes. I have tested on runbot, this module does not cover multiple changes. IMO, similar result can be achieve using auditlog module.

IMO, amendment document should be created first. When amendment approved, it will change contract value.

Reply:
:| all kind of change at the same time on contract are save in this module. 

you got list of change in the amendment ? 

Only the fields note is'nt write if it is alone in vals 
i don't uderstand what you claim on ?